mlrun/runtimes/base.py

```diff
@@ -600,12 +601,28 @@ def _update_run_state(
         updates = None
         last_state = get_in(resp, "status.state", "")
         kind = get_in(resp, "metadata.labels.kind", "")
-        if last_state == "error" or err:
+        if (
+            last_state in mlrun.common.runtimes.constants.RunStates.error_states()
+            or err
+        ):
+            new_state = mlrun.common.runtimes.constants.RunStates.error
+            status_text = None
+            max_retries = get_in(resp, "spec.retry.count", 0)
+            retry_count = get_in(resp, "status.retry_count", 0) or 0
+            if max_retries and retry_count < max_retries:
+                new_state = mlrun.common.runtimes.constants.RunStates.pending_retry
+                status_text = f"Run failed attempt {retry_count + 1} of {max_retries}"
+            elif max_retries and retry_count >= max_retries:
+                status_text = f"Run failed after {retry_count} attempts"
```

Comment:
minor and feel free to ignore, but seems to be a bit more human-readable this way
```suggestion
            if max_retries:
              if retry_count < max_retries:
                  new_state = run_states.pending_retry
                  status_text = f"Run failed attempt {retry_count + 1} of {max_retries}"
              else:
                  status_text = f"Run failed after {retry_count} attempts"
```